scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,874 @@
+
+    def __exit__(self, exception_type, exception_value, traceback) -> None:
+        self.process.kill()
+        self.process.wait(timeout=2.0)
+
+    def trace(self, topic: str, message: str) -> None:
+        if self.trace_io:
+            print(f"{SGR_TRACE}{topic}:{SGR_RESET} {message}")
+
+    def receive_message(self) -> Union[None, dict]:
+        # Note, we should make use of timeout to avoid infinite blocking if nothing is received.
+        CONTENT_LENGTH_HEADER = "Content-Length: "
+        CONTENT_TYPE_HEADER = "Content-Type: "
+        if self.process.stdout == None:
+            return None
+        message_size = None
+        while True:
+            # read header
+            line = self.process.stdout.readline()
+            if line == '':
+                # server quit
+                return None
+            line = line.decode("utf-8")
+            if not line.endswith("\r\n"):
+                raise BadHeader("missing newline")
+            # remove the "\r\n"
+            line = line[:-2]
+            if line == '':
+                break # done with the headers
+            if line.startswith(CONTENT_LENGTH_HEADER):
+                line = line[len(CONTENT_LENGTH_HEADER):]
+                if not line.isdigit():
+                    raise BadHeader("size is not int")
+                message_size = int(line)
+            elif line.startswith(CONTENT_TYPE_HEADER):
+                # nothing todo with type for now.
+                pass
+            else:
+                raise BadHeader("unknown header")
+        if message_size is None:
+            raise BadHeader("missing size")
+        rpc_message = self.process.stdout.read(message_size).decode("utf-8")
+        json_object = json.loads(rpc_message)
+        self.trace('receive_message', json.dumps(json_object, indent=4, sort_keys=True))
+        return json_object
+
+    def send_message(self, method_name: str, params: Optional[dict]) -> None:
+        if self.process.stdin == None:
+            return
+        message = {
+            'jsonrpc': '2.0',
+            'method': method_name,
+            'params': params
+        }
+        json_string = json.dumps(obj=message)
+        rpc_message = f"Content-Length: {len(json_string)}\r\n\r\n{json_string}"
+        self.trace(f'send_message ({method_name})', json.dumps(message, indent=4, sort_keys=True))
+        self.process.stdin.write(rpc_message.encode("utf-8"))
+        self.process.stdin.flush()
+
+    def call_method(self, method_name: str, params: Optional[dict]) -> Any:
+        self.send_message(method_name, params)
+        return self.receive_message()
+
+    def send_notification(self, name: str, params: Optional[dict] = None) -> None:
+        self.send_message(name, params)
+
+# }}}
+
+SGR_RESET = '\033[m'
+SGR_TRACE = '\033[1;36m'
+SGR_NOTICE = '\033[1;35m'
+SGR_TEST_BEGIN = '\033[1;33m'
+SGR_ASSERT_BEGIN = '\033[1;34m'
+SGR_STATUS_OKAY = '\033[1;32m'
+SGR_STATUS_FAIL = '\033[1;31m'
```

Comment:
I think you could now replace these with `Fore`, `Back` and `Style` constants from colorama. That was part of the reason to switch to it after all.

Reply:
The reason to use colorama for me was to actually make this compatible with Windows platform where no VT processing (Windows below version 10 (18H-something) was available. One could use names here, but It's more then foreground, the `1` means "bold", actually :)

Reply:
This will all be removed when we move to pytest.

Reply:
> The reason to use colorama for me was to actually make this compatible with Windows platform

I mean this was another reason but for me it's also important that using colors is simple - I'd like to start using them more in our Python scripts and having to always repeat such a block of definitions would be too annoying and error-prone.

> One could use names here, but It's more then foreground, the 1 means "bold", actually :)

Oh, right. I just noticed that your codes are semantic and not just simple color definitions. I guess it should still be possible to somehow map them to combinations of constants from colorama?

> This will all be removed when we move to pytest.

True. This can be refactored while we move to a test framework.

But you mean unittest of course? pytest is a completely different test framework and I personally don't like it that much (too much magic :)).